designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/scenarioTesting/ResultsWithCountsDtoCodecs.scala

```diff
@@ -22,10 +26,18 @@ object ResultsWithCountsDtoCodecs {
 
   private implicit val testResultsEncoder: Encoder[TestResultsDto] = new Encoder[TestResultsDto]() {
 
-    implicit val nodeResult: Encoder[ResultContext[Json]]                              = deriveConfiguredEncoder
-    implicit val expressionInvocationResult: Encoder[ExpressionInvocationResult[Json]] = deriveConfiguredEncoder
-    implicit val externalInvocationResult: Encoder[ExternalInvocationResult[Json]]     = deriveConfiguredEncoder
-    implicit val nodeTransitionResult: Encoder[NodeTransitionResult]                   = deriveConfiguredEncoder
+    implicit val contextId: Encoder[ContextId] = Encoder.encodeString.contramap(_.legacySerialized)
```

Comment:
We should keep legacy solution as easy as possible to remove. Currently, our base context id encoding strategy, encode context id in legacy format and if you want to use a new approach, you have to use specific `encoderWithDetailedContextId`. Let's invert this relation. For places where we need a legacy contextid format, let's use some explicit legacy encoder, and the default encoder let's encode it to parts. After all, removal of legacy solution should need a few lines marked with TODO to be removed.

Reply:
Inverted, now the legacy value is added by "decorating" the encoder.